src/messages/MessageElement.hpp

```diff
@@ -183,10 +186,14 @@ class MessageElement
     virtual void addToContainer(MessageLayoutContainer &container,
                                 MessageElementFlags flags) = 0;
 
+    virtual QJsonObject toJson() const = 0;
```

Comment:
We could make toJson non-pure and have derived classes call the base toJson instead of jsonBase